snapshot/src/main/java/io/zeebe/snapshots/broker/impl/FileBasedSnapshotStoreFactory.java

```diff
@@ -43,7 +38,14 @@ public PersistedSnapshotStore createSnapshotStore(final Path root, final String
     IoUtil.ensureDirectoryExists(snapshotDirectory.toFile(), "Snapshot directory");
     IoUtil.ensureDirectoryExists(pendingDirectory.toFile(), "Pending snapshot directory");
 
-    return new FileBasedSnapshotStore(
-        new SnapshotMetrics(partitionName), snapshotDirectory, pendingDirectory);
+    return partitionSnapshotStores.computeIfAbsent(
+        partitionName,
+        p ->
+            new FileBasedSnapshotStore(
+                new SnapshotMetrics(partitionName), snapshotDirectory, pendingDirectory));
+  }
+
+  public FileBasedSnapshotStore getSnapshotStore(final String partitionName) {
```

Comment:
I thought we want to have two factory methods in this factory one to create a store which can receive chunks and one which can be used to take snapshots?

Reply:
We will create two factory Interfaces  - `ConstructableStoreFactory` and `ReceivableStoreFactory`